x/liquidstakeibc/keeper/delegation_strategy.go

```diff
@@ -12,34 +12,59 @@ import (
 
 type DelegateAmount struct {
 	ValAddress string
-	ValWeight  sdk.Dec
 	Amount     sdk.Dec
 }
 
 func (k *Keeper) GenerateDelegateMessages(hc *types.HostChain, depositAmount math.Int) ([]proto.Message, error) {
-	return k.generateMessages(hc, depositAmount, false)
+	// filter out validators which are non-delegable (which reached any LSM cap)
+	delegableValidators := make([]*types.Validator, 0)
+	nonDelegableWeight := sdk.ZeroDec()
+	nonDelegableDelegations := sdk.ZeroInt()
+	for _, validator := range hc.Validators {
+		if validator.Delegable {
+			delegableValidators = append(delegableValidators, validator)
+		} else {
+			nonDelegableWeight = nonDelegableWeight.Add(validator.Weight)
+			nonDelegableDelegations = nonDelegableDelegations.Add(validator.DelegatedAmount)
+		}
+	}
+
+	// if there are no delegable validators, do nothing
```

Comment:
actually returning error that may fail some flow, maybe replace "do nothing" comment, or consider returning `nil` error

Reply:
returning an error here just ends the delegation workflow, which will be picked up the next block